src/main/java/com/google/devtools/build/lib/remote/RemoteActionFileSystem.java

```diff
@@ -438,4 +466,84 @@ protected void createHardLink(PathFragment linkPath, PathFragment originalPath)
       throws IOException {
     super.createHardLink(linkPath, originalPath);
   }
+
+  static class RemoteInMemoryFileSystem extends InMemoryFileSystem {
+
+    public RemoteInMemoryFileSystem(DigestHashFunction hashFunction) {
+      super(hashFunction);
+    }
+
+
+    @Override
+    protected FileInfo newFile(Clock clock, PathFragment path) {
+      return new RemoteFileInfo(clock);
+    }
+
+    void injectRemoteFile(PathFragment path, byte[] digest, long size, String actionId)
+        throws IOException {
+      createDirectoryAndParents(path.getParentDirectory());
+      InMemoryContentInfo node = getOrCreateWritableInode(path);
+      if (!(node instanceof RemoteFileInfo)) {
```

Comment:
Could you add a comment as to when this would be the case? My understanding of `InMemoryFileSystem` is that this would happen if a directory or symlink already exists at this location - is that correct?

Reply:
Done.